packages/addons-inject/tests/wrangler.test.js

```diff
@@ -0,0 +1,256 @@
+      .reply(200, "console.log(true);", {
+        headers: {
+          "Content-type": "application/javascript",
+          "X-RTD-Force-Addons": true,
+          "X-RTD-Hosting-Integrations": true,
+        },
+      });
+    let response = await SELF.fetch(
+      "http://test-builds.devthedocs.org/en/latest/_static/foo.js",
+    );
+    expect(response.status).toBe(200);
+    expect(await response.text()).toBe("console.log(true);");
+    expect(response.headers.get("Content-type")).toBe("application/javascript");
+  });
+
+  it("handles searchtools.js", async () => {
+    fetchMock
+      .get("http://test-builds.devthedocs.org")
+      .intercept({ path: "/en/latest/_static/searchtools.js" })
+      .reply(200, AddonsConstants.replacements.searchtools.pattern, {
+        headers: {
+          "Content-type": "application/javascript",
+          "X-RTD-Force-Addons": true,
+          "X-RTD-Hosting-Integrations": true,
+          "X-RTD-Test-Passthrough": 42,
+        },
+      });
+    let response = await SELF.fetch(
+      "http://test-builds.devthedocs.org/en/latest/_static/searchtools.js",
+    );
+    expect(response.status).toBe(200);
+    expect(response.headers.get("Content-type")).toBe("application/javascript");
+    expect(response.headers.get("X-RTD-Test-Passthrough")).toBe("42");
+    expect(await response.text()).toContain(
+      AddonsConstants.replacements.searchtools.replacement,
+    );
+  });
+
+  it("handles error gracefully", async () => {
+    fetchMock
+      .get("http://test-builds.devthedocs.org")
+      .intercept({ path: "/en/latest/" })
+      .reply(200, `<html><head></head><body></body></html>`, {
+        headers: {
+          "Content-type": "text/html",
+          "X-RTD-Force-Addons": true,
+          "X-RTD-Hosting-Integrations": false,
+          "X-RTD-Test-Passthrough": 42,
+          "X-RTD-Debug-Throw-Error": true,
+        },
+      });
+    let response = await SELF.fetch(
+      "http://test-builds.devthedocs.org/en/latest/",
+    );
+    expect(response.status).toBe(200);
+    // Expect the original response and headers
+    expect(await response.text()).toBe(
+      `<html><head></head><body></body></html>`,
+    );
+    expect(response.headers.get("X-RTD-Test-Passthrough")).toBe("42");
+  });
+
+  it("handles binary content", async () => {
+    // It's not clear if this test is helpful. Some operations that were
+    // previously using `Response.text()` were choking on the binary data. This
+    // test may not be catching those points anymore.
+    const binaryData = Buffer.from([
+      137, 80, 78, 71, 13, 10, 26, 10, 0, 0, 0, 13, 73, 72, 68, 82,
+    ]);
+    fetchMock
+      .get("http://test-builds.devthedocs.org")
+      .intercept({ path: "/en/latest/_/static/images/test.png" })
+      .reply(200, binaryData, {
+        headers: {
+          "Content-type": "image/png",
+          "X-RTD-Force-Addons": true,
+          "X-RTD-Hosting-Integrations": false,
```

Comment:
I didn't see any test for `X-RTD-Hosting-Integrations: true`, we should add some if we are using that chunk of code. We can open an issue for this if it's outside the scope of this PR, tho.

Reply:
The issue here is that the logic we'd be testing doesn't seem to quite match what we're doing now. I'm a little confused by the variables/headers used here though.

I noted more here:
https://github.com/readthedocs/common/pull/227/files#r1819843063

I'll add a quick test, but I think that logic sounds wrong now.